### DISCUSSION TITLE
Cannot Link to the callbackUrl

### DISCUSSION THREAD

**Discussion by ThibautMarechal** [sergiodxa/remix-auth-oauth2#32] [Q&A] [answered]
Hello,
I may have missunderstood the concept, but there is this line that prevents me to have a link to the page responsible of the oauth login. (https://github.com/sergiodxa/remix-auth-oauth2/blob/main/src/index.ts#L159)
Is it a bug or I shouldn't have a link to the callbackUrl to trigger the oauth login ?

I wanted to have multiple strategies in one app, (remix-auth-form & remix-auth-oauth2). Bellow the "basic" auth form, a link to oauth callbackUrl.

**Comment by sergiodxa** [comment, marked as answer]
The callback url is to where the Identity Provider will redirect the user after authorizing the login.
This is not to where you take the user to start the authorization process.

You should have a login route with a Sign In button per provider, then make those buttons take the user to a route where you call the authorize method, this should be different than the callback url, from this the strategy will take the user to the identity provider authorization endpoint and then redirect them back to the callback url of your app.

Check the instructions on https://github.com/sergiodxa/remix-auth-github to see how it’s used, that strategy extends the OAuth2Strategy.
